plugins/azure-devops-backend/src/api/AzureDevOpsApi.ts

```diff
@@ -49,23 +47,92 @@ import {
 import { TeamMember as AdoTeamMember } from 'azure-devops-node-api/interfaces/common/VSSInterfaces';
 import { Logger } from 'winston';
 import { PolicyEvaluationRecord } from 'azure-devops-node-api/interfaces/PolicyInterfaces';
-import { WebApi } from 'azure-devops-node-api';
+import {
+  WebApi,
+  getHandlerFromToken,
+  getPersonalAccessTokenHandler,
+} from 'azure-devops-node-api';
 import {
   TeamProjectReference,
   WebApiTeam,
 } from 'azure-devops-node-api/interfaces/CoreInterfaces';
 import { UrlReader } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  AzureDevOpsCredentialsProvider,
+  DefaultAzureDevOpsCredentialsProvider,
+  ScmIntegrations,
+} from '@backstage/integration';
+import {
+  mappedBuildRun,
+  mappedGitTag,
+  mappedPullRequest,
+  mappedRepoBuild,
+} from './mappers';
 
 /** @public */
 export class AzureDevOpsApi {
-  public constructor(
-    private readonly logger: Logger,
-    private readonly webApi: WebApi,
-    private readonly urlReader: UrlReader,
-  ) {}
+  private readonly logger: Logger;
+  private readonly urlReader: UrlReader;
+  private readonly config: Config;
+  private readonly credentialsProvider: AzureDevOpsCredentialsProvider;
+
+  private constructor(
+    logger: Logger,
+    urlReader: UrlReader,
+    config: Config,
+    credentialsProvider: AzureDevOpsCredentialsProvider,
+  ) {
+    this.logger = logger;
+    this.urlReader = urlReader;
+    this.config = config;
+    this.credentialsProvider = credentialsProvider;
+  }
+
+  static fromConfig(
+    config: Config,
+    options: { logger: Logger; urlReader: UrlReader },
+  ) {
+    const scmIntegrations = ScmIntegrations.fromConfig(config);
+    const credentialsProvider =
+      DefaultAzureDevOpsCredentialsProvider.fromIntegrations(scmIntegrations);
+    return new AzureDevOpsApi(
+      options.logger,
+      options.urlReader,
+      config,
+      credentialsProvider,
+    );
+  }
+
+  private async getWebApi(host?: string, org?: string): Promise<WebApi> {
```

Comment:
Nobody calls this ever with a given host and org right? i presume there's some plan here

Reply:
Yes, the plan to tackle adding multi-org support was to do it in chunks - frontend, then backend which is this PR, and then a last PR to wire this all up between. This has made it easier to focus on the specific changes and made the PRs smaller